src/InlineParser.ts

```diff
@@ -0,0 +1,137 @@
+import { InlineElement, InlineGroup } from "./ast";
+import { defaultSchema, Schema, Sugar } from "./schema";
+import { escapeRegExp, last } from "./utils";
+
+interface InlineSyntax {
+	separator?: string;
+	end: string;
+}
+
+export class InlineParser {
+	private readonly regex: RegExp;
+	private readonly inlineTag: InlineSyntax = { separator: "][", end: "]" };
+	private readonly sugars: Map<string, Sugar>;
+	private stack: Array<{ element: InlineElement; syntax: InlineSyntax }> = [];
```

Comment:
I would rename this to `openInlinesStack`. Or perhaps we could define a `type Stack<T> = Array<T>` to syntactically show that it's a stack?

Reply:
Actually, I'm fine with this. The comment I added is enough context.